clippy_utils/src/msrvs.rs

```diff
@@ -58,6 +58,7 @@ msrv_aliases! {
     1,35,0 { OPTION_COPIED, RANGE_CONTAINS }
     1,34,0 { TRY_FROM }
     1,33,0 { UNDERSCORE_IMPORTS }
+    1,31,0 { OPTION_REPLACE }
```

Comment:
This only applies outside of const contexts, the function was const-stabilized in 1.83.0.

So we need a const check with the Marv check.

Reply:
I don't think we do: `mem::replace()` was const-stabilized in 1.83.0 as well, so it will never be found in `const` contexts before 1.83.

Reply:
Then we should document that fact.

Reply:
Done.